octue/cloud/storage/client.py

```diff
@@ -46,17 +47,21 @@ def create_bucket(self, name, location=None, allow_existing=False, timeout=_DEFA
 
         self.client.create_bucket(bucket_or_name=name, location=location, timeout=timeout)
 
-    def upload_file(self, local_path, bucket_name, path_in_bucket, metadata=None, timeout=_DEFAULT_TIMEOUT):
-        """Upload a local file to a Google Cloud bucket at gs://<bucket_name>/<path_in_bucket>.
+    def upload_file(
+        self, local_path, gs_path=None, bucket_name=None, path_in_bucket=None, metadata=None, timeout=_DEFAULT_TIMEOUT
```

Comment:
I think gs_path should be called cloud_path for future API compatibility with other platforms